openwisp_controller/connection/tests/test_models.py

```diff
@@ -563,6 +563,22 @@ def test_command_validation(self):
                     ],
                 )
 
+        with self.subTest('Test command creation without device connection'):
+            device = self._create_device(
+                name='default.test.device2', mac_address='11:22:33:44:55:66'
+            )
+            new_command = Command(
+                device=device, type='custom', input={'command': 'echo test'}
+            )
```

Comment:
Instead of creating new objects, can you we please do this? 

```python
device.deviceconnection_set.delete()
command.type = 'custom'
command.input = {'command': 'echo test'}
with self.assertRaises(ValidationError) as context_manager:
                command.full_clean()
```

This is more of a personal preference. And. this keeps the entire test consistent.

Reply:
With this we need to make sure that this subTest is executed last, if any new subTests are added here in future, as then those subtests will raise error. Just a heads up

Reply:
It should be okay, I guess.